Automattic/pocket-casts-android
# Add pager to watch app

## Description
This adds a three screen pager to the watch app. The first page will be the current page. The second page will always be the now playing screen. The third page will always be the Up Next screen.

I initially wanted to create a `Pager` at the top level and have navigation within the first page, but it looks like that [isn't really possible currently](https://github.com/google/horologist/issues/1181).

## Testing Instructions

Verify that the pager works as expected and is present on the screens you would expect. Transient notification-style screens and the Settings screen are the only ones that should not have the pager. Also check that back navigation works as expected (i.e., swiping back and swiping to exit the app).

Note that you should only be able to swipe back to the previous screen if you are on the `0` page of the pager. I did this to avoid having users tap on the episode title in the Now Playing screen, swipe to the Now Playing screen again from the episode screen, and then having swipe to dismiss take them from one Now Playing screen to their previous (identical) Now Playing screen on the previous pager.

Lastly, this PR includes a somewhat-unrelated change: Tapping on the podcast name on the episode details screen should now take you to the podcast screen. (internal discussion: p1682360221894429/1682357294.725699-slack-C040S9BK7SP)

## Screenshots or Screencast 

https://user-images.githubusercontent.com/4656348/234282129-936e12c1-e1fd-41d9-9d31-559cdb1c02b7.mov

## Checklist
